Modix.Data/Repositories/MessageRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Modix.Data.Models.Core;
+
+namespace Modix.Data.Repositories
+{
+    public interface IMessageRepository
+    {
+        Task<IReadOnlyCollection<MessageEntity>> GetRecentUserMessagesAsync(ulong guildId, ulong userId, TimeSpan timespan);
```

Comment:
Entity classes should not be exposed outside of `Modix.Data`. Plus, this would require multiple separate queries to translate AuthorId, ChannelId, etc. into meaningful informatoin. Write a projection class to represent the data that is actually needed.

Also, `userId` should be `authorId` to match the field on the entity that you're filtering against.

Also, this would be perfectly suited for a `MessageSearchCriteria` class.